### DISCUSSION TITLE
Making a worldwide custom provider

### DISCUSSION THREAD

**Discussion by mrbellek** [azuyalabs/yasumi#353] [Q&A]
I'm looking to fetch all holidays worldwide for a given single day. Is this possible using a custom provider, or am I better off running through all countries and adding up any active holidays?

**Comment by stelgenhof** [comment]
@mrbellek A custom provider is more useful when you have a need to customize an existing provider (country). You could make it work with a custom provider, however I'd recommend running through all providers and check for any holiday the day you are interested in.

**Reply by mrbellek** [reply]
Thank you!

I also ran across the problem that 'international' unofficial holidays aren't in the library, such as [International Day of Persons with Disabilities](https://nationaltoday.com/international-day-of-persons-with-disabilities/). Is it possible to make a custom provider for that, like a provider for the fictional country 'World'?

**Comment by stelgenhof** [comment]
@mrbellek Yes, Yasumi concentrates primarily on official holidays (recognized by global and local jurisdictions, etc.) that in some form also mean a non-working day. Having said that, there are exceptions :) Some providers do include non-official holidays such as "Valentine's Day", "World Animal Day", etc. Yasumi is 'free' (in terms of limits) and non-official or "international holidays" can certainly be added to a provider if that is wanted.

If your goal is to have all those "international" / unofficial holidays gathered under one wing, then you could indeed create a custom provider for that.

**Reply by mrbellek** [reply]
Excellent, thank you for your help.

**Reply by stelgenhof** [reply]
You're welcome! If there's any other questions, don't hesitate to use GitHub's discussions here :)
